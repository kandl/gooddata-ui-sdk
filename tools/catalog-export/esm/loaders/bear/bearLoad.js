// (C) 2007-2023 GoodData Corporation
import ora from "ora";
import { logError } from "../../cli/loggers.js";
import { CatalogExportError } from "../../base/types.js";
import { loadCatalog } from "./bearCatalog.js";
import { loadDateDataSets } from "./bearDateDatasets.js";
import { loadInsights } from "./bearInsights.js";
import { loadAnalyticalDashboard } from "./bearAnalyticalDashboard.js";
/**
 * Loads all workspace metadata that can be used for exporting into catalog.
 *
 * @param workspaceId - workspace identifier
 * @throws CatalogExportError
 */
export async function bearLoad(workspaceId) {
    var _a;
    const spinner = ora();
    try {
        spinner.start("Loading catalog of attributes and metrics…");
        const catalog = await loadCatalog(workspaceId);
        spinner.succeed("Catalog loaded");
        spinner.start("Loading date data sets…");
        const dateDataSets = await loadDateDataSets(workspaceId);
        spinner.succeed("Date data sets loaded");
        spinner.start("Loading insights…");
        const insights = await loadInsights(workspaceId);
        spinner.succeed("Insights loaded");
        spinner.start("Loading analytical dashboards…");
        const analyticalDashboards = await loadAnalyticalDashboard(workspaceId);
        spinner.succeed("Analytical dashboards loaded");
        return {
            workspaceId,
            catalog,
            dateDataSets,
            insights,
            analyticalDashboards,
        };
    }
    catch (err) {
        spinner.fail();
        if (((_a = err === null || err === void 0 ? void 0 : err.response) === null || _a === void 0 ? void 0 : _a.status) === 404) {
            // handle known error more gracefully to avoid general-type error messages
            logError(`Workspace with id '${workspaceId}' was not found. Please make sure you are passing a correct value to the 'workspace-id' argument.`);
            process.exit(1);
        }
        const more = err.response ? err.response.url : "";
        logError(`Exception: ${err.message} ${more}\n${err.stack}`);
        throw new CatalogExportError("A fatal error has occurred while loading project metadata", 1);
    }
}
