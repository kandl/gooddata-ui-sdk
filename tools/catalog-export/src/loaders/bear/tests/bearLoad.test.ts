// (C) 2007-2021 GoodData Corporation
import { bearLoad } from "../bearLoad.js";

describe("bearLoad", () => {
    it("should transfer project ID", async () => {
        const result = await bearLoad("test");

        expect(result.workspaceId).toEqual("test");
    });

    it("should load ldm", async () => {
        const result = await bearLoad("test");

        expect(result.catalog).toMatchSnapshot();
    });

    it("should load date data sets", async () => {
        const result = await bearLoad("test");

        expect(result.dateDataSets).toMatchSnapshot();
    });

    it("should load visualizations", async () => {
        const result = await bearLoad("test");

        expect(result.insights).toMatchSnapshot();
    });

    it("should load analytical dashboards", async () => {
        const result = await bearLoad("test");

        expect(result.analyticalDashboards).toMatchSnapshot();
    });
});
