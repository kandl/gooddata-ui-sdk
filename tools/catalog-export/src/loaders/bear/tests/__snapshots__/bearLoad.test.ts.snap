// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`bearLoad > should load analytical dashboards 1`] = `
[
  {
    "identifier": "afMA17GSbk31",
    "tags": "",
    "title": "KPIs",
  },
  {
    "identifier": "abBJlnxrfEWH",
    "tags": "",
    "title": "Store management KPIs",
  },
  {
    "identifier": "aahlrtUEgKkq",
    "tags": "",
    "title": "Untitled",
  },
]
`;

exports[`bearLoad > should load date data sets 1`] = `
[
  {
    "dateDataSet": {
      "content": {
        "attributes": [
          {
            "attribute": {
              "content": {
                "displayForms": [
                  {
                    "meta": {
                      "identifier": "timeline.aag81lMifn6q",
                      "tags": "",
                      "title": "Year (Timeline)",
                    },
                  },
                ],
              },
              "meta": {
                "identifier": "timeline.year",
                "tags": "date year",
                "title": "Year (Timeline)",
              },
            },
          },
          {
            "attribute": {
              "content": {
                "displayForms": [
                  {
                    "meta": {
                      "identifier": "timeline.aam81lMifn6q",
                      "tags": "",
                      "title": "default (Timeline)",
                    },
                  },
                ],
              },
              "meta": {
                "identifier": "timeline.quarter.in.year",
                "tags": "date quarter year",
                "title": "Quarter (Timeline)",
              },
            },
          },
          {
            "attribute": {
              "content": {
                "displayForms": [
                  {
                    "meta": {
                      "identifier": "timeline.aaA81lMifn6q",
                      "tags": "",
                      "title": "Week #/Year (W1/2010) (Timeline)",
                    },
                  },
                  {
                    "meta": {
                      "identifier": "timeline.aaw81lMifn6q",
                      "tags": "",
                      "title": "Week Starting (Timeline)",
                    },
                  },
                  {
                    "meta": {
                      "identifier": "timeline.aau81lMifn6q",
                      "tags": "",
                      "title": "From - To (Timeline)",
                    },
                  },
                  {
                    "meta": {
                      "identifier": "timeline.aay81lMifn6q",
                      "tags": "",
                      "title": "Week #/Year (Cont.) (Timeline)",
                    },
                  },
                  {
                    "meta": {
                      "identifier": "timeline.aaC81lMifn6q",
                      "tags": "",
                      "title": "Wk/Qtr/Year (Cont.) (Timeline)",
                    },
                  },
                  {
                    "meta": {
                      "identifier": "timeline.aas81lMifn6q",
                      "tags": "",
                      "title": "Wk/Qtr/Year (Timeline)",
                    },
                  },
                ],
              },
              "meta": {
                "identifier": "timeline.week",
                "tags": "date week",
                "title": "Week (Sun-Sat)/Year (Timeline)",
              },
            },
          },
          {
            "attribute": {
              "content": {
                "displayForms": [
                  {
                    "meta": {
                      "identifier": "timeline.aaI81lMifn6q",
                      "tags": "",
                      "title": "Number US (Timeline)",
                    },
                  },
                ],
              },
              "meta": {
                "identifier": "timeline.week.in.year",
                "tags": "date week year",
                "title": "Week (Sun-Sat) (Timeline)",
              },
            },
          },
          {
            "attribute": {
              "content": {
                "displayForms": [
                  {
                    "meta": {
                      "identifier": "timeline.aaO81lMifn6q",
                      "tags": "",
                      "title": "Number US (Timeline)",
                    },
                  },
                ],
              },
              "meta": {
                "identifier": "timeline.week.in.quarter",
                "tags": "date quarter week",
                "title": "Week (Sun-Sat) of Qtr (Timeline)",
              },
            },
          },
          {
            "attribute": {
              "content": {
                "displayForms": [
                  {
                    "meta": {
                      "identifier": "timeline.aa281lMifn6q",
                      "tags": "",
                      "title": "Week #/Year (W1/2010) (Timeline)",
                    },
                  },
                  {
                    "meta": {
                      "identifier": "timeline.aaY81lMifn6q",
                      "tags": "",
                      "title": "Week Starting (Timeline)",
                    },
                  },
                  {
                    "meta": {
                      "identifier": "timeline.aaW81lMifn6q",
                      "tags": "",
                      "title": "From - To (Timeline)",
                    },
                  },
                ],
              },
              "meta": {
                "identifier": "timeline.euweek",
                "tags": "date eu week",
                "title": "Week (Mon-Sun)/Year (Timeline)",
              },
            },
          },
          {
            "attribute": {
              "content": {
                "displayForms": [
                  {
                    "meta": {
                      "identifier": "timeline.aba81lMifn6q",
                      "tags": "",
                      "title": "Number EU (Timeline)",
                    },
                  },
                ],
              },
              "meta": {
                "identifier": "timeline.euweek.in.year",
                "tags": "date eu week year",
                "title": "Week (Mon-Sun) (Timeline)",
              },
            },
          },
          {
            "attribute": {
              "content": {
                "displayForms": [
                  {
                    "meta": {
                      "identifier": "timeline.abg81lMifn6q",
                      "tags": "",
                      "title": "Number EU (Timeline)",
                    },
                  },
                ],
              },
              "meta": {
                "identifier": "timeline.euweek.in.quarter",
                "tags": "date eu quarter week",
                "title": "Week (Mon-Sun) of Qtr (Timeline)",
              },
            },
          },
          {
            "attribute": {
              "content": {
                "displayForms": [
                  {
                    "meta": {
                      "identifier": "timeline.abm81lMifn6q",
                      "tags": "",
                      "title": "Short (Jan) (Timeline)",
                    },
                  },
                  {
                    "meta": {
                      "identifier": "timeline.abs81lMifn6q",
                      "tags": "",
                      "title": "Long (January) (Timeline)",
                    },
                  },
                  {
                    "meta": {
                      "identifier": "timeline.abq81lMifn6q",
                      "tags": "",
                      "title": "Number (M1) (Timeline)",
                    },
                  },
                  {
                    "meta": {
                      "identifier": "timeline.abo81lMifn6q",
                      "tags": "",
                      "title": "M/Q (M1/Q1) (Timeline)",
                    },
                  },
                ],
              },
              "meta": {
                "identifier": "timeline.month.in.year",
                "tags": "date month year",
                "title": "Month (Timeline)",
              },
            },
          },
          {
            "attribute": {
              "content": {
                "displayForms": [
                  {
                    "meta": {
                      "identifier": "timeline.aby81lMifn6q",
                      "tags": "",
                      "title": "Number (Timeline)",
                    },
                  },
                ],
              },
              "meta": {
                "identifier": "timeline.month.in.quarter",
                "tags": "date month quarter",
                "title": "Month of Quarter (Timeline)",
              },
            },
          },
          {
            "attribute": {
              "content": {
                "displayForms": [
                  {
                    "meta": {
                      "identifier": "timeline.abE81lMifn6q",
                      "tags": "",
                      "title": "default (Timeline)",
                    },
                  },
                ],
              },
              "meta": {
                "identifier": "timeline.day.in.year",
                "tags": "date day year",
                "title": "Day of Year (Timeline)",
              },
            },
          },
          {
            "attribute": {
              "content": {
                "displayForms": [
                  {
                    "meta": {
                      "identifier": "timeline.abK81lMifn6q",
                      "tags": "",
                      "title": "Short (Sun) (Timeline)",
                    },
                  },
                  {
                    "meta": {
                      "identifier": "timeline.abO81lMifn6q",
                      "tags": "",
                      "title": "Long (Sunday) (Timeline)",
                    },
                  },
                  {
                    "meta": {
                      "identifier": "timeline.abM81lMifn6q",
                      "tags": "",
                      "title": "Number (1=Sunday) (Timeline)",
                    },
                  },
                ],
              },
              "meta": {
                "identifier": "timeline.day.in.week",
                "tags": "date day week",
                "title": "Day of Week (Sun-Sat) (Timeline)",
              },
            },
          },
          {
            "attribute": {
              "content": {
                "displayForms": [
                  {
                    "meta": {
                      "identifier": "timeline.abU81lMifn6q",
                      "tags": "",
                      "title": "Short (Mon) (Timeline)",
                    },
                  },
                  {
                    "meta": {
                      "identifier": "timeline.abY81lMifn6q",
                      "tags": "",
                      "title": "Long (Monday) (Timeline)",
                    },
                  },
                  {
                    "meta": {
                      "identifier": "timeline.abW81lMifn6q",
                      "tags": "",
                      "title": "Number (1=Monday) (Timeline)",
                    },
                  },
                ],
              },
              "meta": {
                "identifier": "timeline.day.in.euweek",
                "tags": "date day eu week",
                "title": "Day of Week (Mon-Sun) (Timeline)",
              },
            },
          },
          {
            "attribute": {
              "content": {
                "displayForms": [
                  {
                    "meta": {
                      "identifier": "timeline.ab481lMifn6q",
                      "tags": "",
                      "title": "default (Timeline)",
                    },
                  },
                ],
              },
              "meta": {
                "identifier": "timeline.day.in.quarter",
                "tags": "date day quarter",
                "title": "Day of Quarter (Timeline)",
              },
            },
          },
          {
            "attribute": {
              "content": {
                "displayForms": [
                  {
                    "meta": {
                      "identifier": "timeline.aca81lMifn6q",
                      "tags": "",
                      "title": "default (Timeline)",
                    },
                  },
                ],
              },
              "meta": {
                "identifier": "timeline.day.in.month",
                "tags": "date day month",
                "title": "Day of Month (Timeline)",
              },
            },
          },
          {
            "attribute": {
              "content": {
                "displayForms": [
                  {
                    "meta": {
                      "identifier": "timeline.aci81lMifn6q",
                      "tags": "",
                      "title": "US Short (Timeline)",
                    },
                  },
                ],
              },
              "meta": {
                "identifier": "timeline.quarter",
                "tags": "date quarter",
                "title": "Quarter/Year (Timeline)",
              },
            },
          },
          {
            "attribute": {
              "content": {
                "displayForms": [
                  {
                    "meta": {
                      "identifier": "timeline.act81lMifn6q",
                      "tags": "",
                      "title": "Short (Jan 2010) (Timeline)",
                    },
                  },
                  {
                    "meta": {
                      "identifier": "timeline.acx81lMifn6q",
                      "tags": "",
                      "title": "Long (January 2010) (Timeline)",
                    },
                  },
                  {
                    "meta": {
                      "identifier": "timeline.acv81lMifn6q",
                      "tags": "",
                      "title": "Number (1/2010) (Timeline)",
                    },
                  },
                ],
              },
              "meta": {
                "identifier": "timeline.month",
                "tags": "date month",
                "title": "Month/Year (Timeline)",
              },
            },
          },
          {
            "attribute": {
              "content": {
                "displayForms": [
                  {
                    "meta": {
                      "identifier": "timeline.date.mmddyyyy",
                      "tags": "",
                      "title": "mm/dd/yyyy (Timeline)",
                    },
                  },
                  {
                    "meta": {
                      "identifier": "timeline.date.yyyymmdd",
                      "tags": "",
                      "title": "yyyy-mm-dd (Timeline)",
                    },
                  },
                  {
                    "meta": {
                      "identifier": "timeline.date.mdyy",
                      "tags": "",
                      "title": "m/d/yy (no leading zeroes) (Timeline)",
                    },
                  },
                  {
                    "meta": {
                      "identifier": "timeline.date.long",
                      "tags": "",
                      "title": "Long (Mon, Jan 1, 2010) (Timeline)",
                    },
                  },
                  {
                    "meta": {
                      "identifier": "timeline.date.ddmmyyyy",
                      "tags": "",
                      "title": "dd/mm/yyyy (Timeline)",
                    },
                  },
                  {
                    "meta": {
                      "identifier": "timeline.date.eddmmyyyy",
                      "tags": "",
                      "title": "dd-mm-yyyy (Timeline)",
                    },
                  },
                ],
              },
              "meta": {
                "identifier": "timeline.date",
                "tags": "date",
                "title": "Date (Timeline)",
              },
            },
          },
        ],
      },
      "meta": {
        "identifier": "timeline.dataset.dt",
        "tags": "",
        "title": "Date (Timeline)",
      },
    },
  },
]
`;

exports[`bearLoad > should load ldm 1`] = `
{
  "attributes": [
    {
      "attribute": {
        "content": {
          "dimension": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/5202",
          "displayForms": [
            {
              "content": {
                "expression": "[/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/5029]",
                "formOf": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/5208",
              },
              "links": {
                "elements": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/5209/elements",
              },
              "meta": {
                "author": "/gdc/account/profile/785e4edc1c8dda4783bcd76147547ed9",
                "category": "attributeDisplayForm",
                "contributor": "/gdc/account/profile/785e4edc1c8dda4783bcd76147547ed9",
                "created": "2019-03-12 10:59:01",
                "deprecated": "0",
                "identifier": "label.orders.channel",
                "isProduction": 1,
                "summary": "",
                "tags": "",
                "title": "Channel",
                "updated": "2019-03-12 10:59:01",
                "uri": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/5209",
              },
            },
          ],
          "fk": [
            {
              "data": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/5014",
              "multivalue": false,
              "type": "col",
            },
          ],
          "pk": [
            {
              "data": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/5027",
              "type": "col",
            },
          ],
        },
        "meta": {
          "author": "/gdc/account/profile/785e4edc1c8dda4783bcd76147547ed9",
          "category": "attribute",
          "contributor": "/gdc/account/profile/785e4edc1c8dda4783bcd76147547ed9",
          "created": "2019-03-12 10:59:01",
          "deprecated": "0",
          "identifier": "attr.orders.channel",
          "isProduction": 1,
          "summary": "",
          "tags": "",
          "title": "Channel",
          "updated": "2019-03-12 10:59:01",
          "uri": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/5208",
        },
      },
    },
    {
      "attribute": {
        "content": {
          "dimension": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/161",
          "displayForms": [
            {
              "content": {
                "default": 1,
                "expression": "[/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/153]",
                "formOf": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/686",
              },
              "links": {
                "elements": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/688/elements",
              },
              "meta": {
                "author": "/gdc/account/profile/8d381a33fff78cf82392c837246d362e",
                "category": "attributeDisplayForm",
                "contributor": "/gdc/account/profile/9bf17dfc423644e9bf92925bab65a3e9",
                "created": "2018-04-27 11:13:08",
                "deprecated": "0",
                "identifier": "label.employee.employeename",
                "isProduction": 1,
                "summary": "",
                "tags": "",
                "title": "Employee Name",
                "updated": "2018-05-15 16:18:26",
                "uri": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/688",
              },
            },
            {
              "content": {
                "expression": "[/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/155]",
                "formOf": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/686",
                "type": "GDC.link",
              },
              "links": {
                "elements": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/687/elements",
              },
              "meta": {
                "author": "/gdc/account/profile/8d381a33fff78cf82392c837246d362e",
                "category": "attributeDisplayForm",
                "contributor": "/gdc/account/profile/9bf17dfc423644e9bf92925bab65a3e9",
                "created": "2018-04-27 11:13:08",
                "deprecated": "0",
                "identifier": "label.employee.employeename.employeeurl",
                "isProduction": 1,
                "summary": "",
                "tags": "",
                "title": "Employee URL",
                "updated": "2018-05-15 16:18:26",
                "uri": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/687",
              },
            },
          ],
          "fk": [
            {
              "data": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/9",
              "multivalue": false,
              "type": "col",
            },
          ],
          "linkAttributeDF": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/687",
          "pk": [
            {
              "data": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/151",
              "type": "col",
            },
          ],
        },
        "meta": {
          "author": "/gdc/account/profile/8d381a33fff78cf82392c837246d362e",
          "category": "attribute",
          "contributor": "/gdc/account/profile/9bf17dfc423644e9bf92925bab65a3e9",
          "created": "2018-04-27 11:13:08",
          "deprecated": "0",
          "identifier": "attr.employee.employeename",
          "isProduction": 1,
          "summary": "",
          "tags": "",
          "title": "Employee Name",
          "updated": "2018-05-15 16:18:26",
          "uri": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/686",
        },
      },
    },
  ],
  "facts": [
    {
      "fact": {
        "content": {
          "expr": [
            {
              "data": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/24",
              "type": "col",
            },
          ],
          "folders": [
            "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/164",
          ],
        },
        "meta": {
          "author": "/gdc/account/profile/8d381a33fff78cf82392c837246d362e",
          "category": "fact",
          "contributor": "/gdc/account/profile/9bf17dfc423644e9bf92925bab65a3e9",
          "created": "2018-04-27 11:13:09",
          "deprecated": "0",
          "identifier": "fact.salesdetailfact.menuitemsales",
          "isProduction": 1,
          "summary": "",
          "tags": "",
          "title": "$ Menu Item Sales",
          "updated": "2018-05-15 16:18:26",
          "uri": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/706",
        },
      },
    },
    {
      "fact": {
        "content": {
          "expr": [
            {
              "data": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/82",
              "type": "col",
            },
          ],
          "folders": [
            "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/163",
          ],
        },
        "meta": {
          "author": "/gdc/account/profile/8d381a33fff78cf82392c837246d362e",
          "category": "fact",
          "contributor": "/gdc/account/profile/9bf17dfc423644e9bf92925bab65a3e9",
          "created": "2018-04-27 11:13:08",
          "deprecated": "0",
          "identifier": "fact.restaurantcostsfact.cost",
          "isProduction": 1,
          "summary": "",
          "tags": "",
          "title": "Cost",
          "updated": "2018-05-15 16:18:26",
          "uri": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/704",
        },
      },
    },
  ],
  "metrics": [
    {
      "metric": {
        "content": {
          "expression": "SELECT COUNT([/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/289])  BY [/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/289], ALL OTHER WHERE [/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/289] BETWEEN [/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/789]-3 AND [/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/789]",
          "format": "#,##0.00",
          "tree": {
            "content": [
              {
                "content": [
                  {
                    "content": [
                      {
                        "position": {
                          "column": 27,
                          "line": 1,
                        },
                        "type": "attribute object",
                        "value": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/289",
                      },
                    ],
                    "position": {
                      "column": 21,
                      "line": 1,
                    },
                    "type": "function",
                    "value": "COUNT",
                  },
                ],
                "position": {
                  "column": 21,
                  "line": 1,
                },
                "type": "expression",
              },
              {
                "content": [
                  {
                    "position": {
                      "column": 84,
                      "line": 1,
                    },
                    "type": "attribute object",
                    "value": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/289",
                  },
                  {
                    "content": [],
                    "position": {
                      "column": 137,
                      "line": 1,
                    },
                    "type": "all other",
                  },
                ],
                "position": {
                  "column": 81,
                  "line": 1,
                },
                "type": "by",
              },
              {
                "content": [
                  {
                    "content": [
                      {
                        "position": {
                          "column": 153,
                          "line": 1,
                        },
                        "type": "attribute object",
                        "value": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/289",
                      },
                      {
                        "content": [
                          {
                            "position": {
                              "column": 213,
                              "line": 1,
                            },
                            "type": "metric object",
                            "value": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/789",
                          },
                          {
                            "position": {
                              "column": 265,
                              "line": 1,
                            },
                            "type": "number",
                            "value": "3",
                          },
                        ],
                        "position": {
                          "column": 264,
                          "line": 1,
                        },
                        "type": "-",
                      },
                      {
                        "position": {
                          "column": 271,
                          "line": 1,
                        },
                        "type": "metric object",
                        "value": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/789",
                      },
                    ],
                    "position": {
                      "column": 205,
                      "line": 1,
                    },
                    "type": "between",
                  },
                ],
                "position": {
                  "column": 147,
                  "line": 1,
                },
                "type": "where",
              },
            ],
            "position": {
              "column": 14,
              "line": 1,
            },
            "type": "metric",
          },
        },
        "links": {
          "explain2": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/867/explain2",
        },
        "meta": {
          "author": "/gdc/account/profile/8d381a33fff78cf82392c837246d362e",
          "category": "metric",
          "contributor": "/gdc/account/profile/9bf17dfc423644e9bf92925bab65a3e9",
          "created": "2018-04-27 14:19:09",
          "deprecated": "0",
          "identifier": "aaIHiWZjfWNA",
          "isProduction": 1,
          "summary": "",
          "tags": "hidden",
          "title": "_Filter Last 4 Quarter",
          "updated": "2018-08-09 14:13:28",
          "uri": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/867",
        },
      },
    },
    {
      "metric": {
        "content": {
          "expression": "SELECT COUNT([/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/289])  BY [/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/289], ALL OTHER WHERE [/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/289] = [/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/789]",
          "folders": [
            "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/788",
          ],
          "format": "#,##0.00",
          "tree": {
            "content": [
              {
                "content": [
                  {
                    "content": [
                      {
                        "position": {
                          "column": 27,
                          "line": 1,
                        },
                        "type": "attribute object",
                        "value": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/289",
                      },
                    ],
                    "position": {
                      "column": 21,
                      "line": 1,
                    },
                    "type": "function",
                    "value": "COUNT",
                  },
                ],
                "position": {
                  "column": 21,
                  "line": 1,
                },
                "type": "expression",
              },
              {
                "content": [
                  {
                    "position": {
                      "column": 84,
                      "line": 1,
                    },
                    "type": "attribute object",
                    "value": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/289",
                  },
                  {
                    "content": [],
                    "position": {
                      "column": 137,
                      "line": 1,
                    },
                    "type": "all other",
                  },
                ],
                "position": {
                  "column": 81,
                  "line": 1,
                },
                "type": "by",
              },
              {
                "content": [
                  {
                    "content": [
                      {
                        "position": {
                          "column": 153,
                          "line": 1,
                        },
                        "type": "attribute object",
                        "value": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/289",
                      },
                      {
                        "position": {
                          "column": 207,
                          "line": 1,
                        },
                        "type": "metric object",
                        "value": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/789",
                      },
                    ],
                    "position": {
                      "column": 205,
                      "line": 1,
                    },
                    "type": "=",
                  },
                ],
                "position": {
                  "column": 147,
                  "line": 1,
                },
                "type": "where",
              },
            ],
            "position": {
              "column": 14,
              "line": 1,
            },
            "type": "metric",
          },
        },
        "links": {
          "explain2": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/880/explain2",
        },
        "meta": {
          "author": "/gdc/account/profile/8d381a33fff78cf82392c837246d362e",
          "category": "metric",
          "contributor": "/gdc/account/profile/ea6a1e8992361e1401d9accc9789cb5a",
          "created": "2018-04-27 14:19:09",
          "deprecated": "0",
          "identifier": "aaiF4bffe4yn",
          "isProduction": 1,
          "summary": "",
          "tags": "hidden",
          "title": "_Filter Quarter",
          "updated": "2018-05-24 09:05:24",
          "uri": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/880",
        },
      },
    },
  ],
}
`;

exports[`bearLoad > should load visualizations 1`] = `
[
  {
    "identifier": "abhJpedgcfU2",
    "tags": "",
    "title": "Labor Costs vs Scheduled Costs",
  },
  {
    "identifier": "acPWDTL2bJeX",
    "tags": "",
    "title": "Total Sales by City",
  },
  {
    "identifier": "aaBN5UG3dXu4",
    "tags": "",
    "title": "$ Total Sales by Location",
  },
]
`;
